lbr.py

```diff
@@ -476,6 +477,19 @@ def prev_line(i=-1):
   if xip in indirects:
     inc(hsts['indirect_%s_targets' % hex_ip(xip)], ip)
     inc(hsts['indirect_%s_paths' % hex_ip(xip)], '%s.%s.%s' % (hex_ip(get_taken(lines, -2)['from']), hex_ip(xip), hex_ip(ip)))
+  mrn_dst=x86.get("dst",line)
+  def mrn_cond(l):return not is_type(x86.JUMP,l) and re.search(x86.MEM_IDX,l)and not re.search("xmm|ymm|zmm|rip",l)
+  if is_type("inc-dec",line) and x86.is_memory(line) and re.search(x86.MEM_IDX,line):
+    inc_stat('%s non-MRNable' % ('INC' if 'inc' in x86.get('inst',line) else 'DEC'))
+  elif mrn_cond(line) and x86.is_mem_store(line) and not re.search("ah|bh|ch|dh",mrn_dst) and not re.search("%rip",line) and re.search("ax|bx|cx|dx|si|di|bp|sp|r8w|r9w|r10w|r11w|r12w|r13w|r14w|r15w",line):
```

Comment:
Considering the requested updates above, you'll need to add another check here x86.is_mem_MRN().
Also, please re-write regexes to minimize duplications e.g. xmm|ymm|zmm -> [x-z]mm, also the very long regex after.
Better replace the <re.search(%rip, line)> with a simple <'%rip' in line>, no need for regex here.